dissect/target/plugins/os/unix/linux/recentlyused.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime
+from typing import Iterator, Union
+
+from defusedxml import ElementTree
+from flow.record import GroupedRecord
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+
+RecentlyUsedRecord = TargetRecordDescriptor(
+    "unix/linux/recentlyused",
+    [
+        ("datetime", "ts"),
+        ("string", "user"),
+        ("string", "source"),
+        ("string", "href"),
+        ("datetime", "added"),
+        ("datetime", "modified"),
+        ("datetime", "visited"),
+        ("string", "mimetype"),
+        ("string", "groups"),
+        ("boolean", "private"),
+    ],
+)
+
+RecentlyUsedIconRecord = TargetRecordDescriptor(
+    "unix/linux/recentlyusedicon",
+    [
+        ("string", "type"),
+        ("string", "href"),
+        ("string", "name"),
+    ],
+)
+
+RecentlyUsedApplicationRecord = TargetRecordDescriptor(
+    "unix/linux/recentlyusedapplication",
+    [
+        ("datetime", "ts"),
+        ("string", "name"),
+        ("string", "exec"),
+        ("varint", "count"),
+    ],
+)
+ns = {
+    "bookmark": "http://www.freedesktop.org/standards/desktop-bookmarks",
+    "mime": "http://www.freedesktop.org/standards/shared-mime-info",
+}
+
+
+def parse_ts(ts):
+    """Parse timestamp format from xbel file"""
+    # datetime.fromisoformat() doesn´t support the trailing Z in python <= 3.10
+    return datetime.strptime(ts, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+
+def parse_recentlyused_xbel(username, xbel_file):
+    with xbel_file.open() as fh:
+        et = ElementTree.fromstring(fh.read(), forbid_dtd=True)
+        for b in et.iter("bookmark"):
+            # The spec says there should always be exactly one.
+            # Ignore if there are fewer or more.
+            mimetypes = b.findall("./info/metadata/mime:mime-type", ns)
+            if mimetypes and len(mimetypes) == 1:
+                mimetype = mimetypes[0].get("type")
+            else:
+                mimetype = None
+
+            # This is just a list of names, GroupedRecords seem overkill
+            groups = b.findall("./info/metadata/bookmark:groups/bookmark:group", ns)
+            group_list = ", ".join(group.text for group in groups)
+
+            # There should be at most one "private" tag, but accept multiple
+            private_entries = b.findall("./info/metadata/bookmark:private", ns)
+            private = private_entries is not None and len(private_entries) > 0
+
+            cur = RecentlyUsedRecord(
+                ts=parse_ts(b.get("visited")),
+                user=username,
+                source=xbel_file,
+                href=b.get("href"),
+                added=parse_ts(b.get("added")),
+                modified=parse_ts(b.get("modified")),
+                visited=parse_ts(b.get("visited")),
+                mimetype=mimetype,
+                groups=group_list,
+                private=private,
```

Comment:
Thanks, TIL.

I'm not sure what type `RecentlyUsedIconRecord` and `RecentlyUsedApplicationRecord` should be then. All other `GroupedRecord`s also seem to be using `TargetRecordDescriptor` or `create_extended_descriptor`. What would you suggest?

Reply:
I ended up still using the TargerRecordDescriptor, but not filling the `_target` field to prevent duplicate data. Does that seem like the right approach to you?